### PR TITLE
Feature/cli nouns list

### PR DESCRIPTION
**Pull request by roblim**
- clean up `great_expectations docs list`, `great_expectations suite list`, and `great_expectations datasource list`
- add `great_expectations store list` and `great_expectations validation-operator list` operations

Screenshot 2020-03-23 00 56 00
Screenshot 2020-03-23 00 55 24

### CODE REVIEW

File: great_expectations/render/renderer/site_builder.py [hunk 1]
```diff
@@ -205,7 +205,8 @@ def build(self, resource_identifiers=None):
         for site_section, site_section_builder in self.site_section_builders.items():
             site_section_builder.build(resource_identifiers=resource_identifiers)
 
-        return self.site_index_builder.build()
+        index_page_resource_idrentifie_tuple = self.site_index_builder.build()
```
Comment (jcampbell) [inline great_expectations/render/renderer/site_builder.py]: Minor --> typo in "idrentifie"

File: great_expectations/data_context/types/base.py [hunk 2]
```diff
@@ -191,7 +190,7 @@ class Meta:
         unknown = INCLUDE
 
     class_name = fields.String(required=True)
-    module_name = fields.String(allow_none=True)
+    module_name = fields.String(missing="great_expectations.datasource")
```
Comment (jcampbell) [inline great_expectations/data_context/types/base.py:193]: I have a preference for adjusting to completely omit the key from the config when it has the default value, and also to ensure we don't have defaults specified in multiple places (there is an existing default set via `instantiate_class_from_config`)

File: tests/cli/test_datasource_pandas.py [hunk 3]
```diff
@@ -53,11 +63,21 @@ def test_cli_datasorce_list(caplog, empty_data_context, filesystem_csv_2):
     result = runner.invoke(
         cli, ["datasource", "list", "-d", project_root_dir], catch_exceptions=False
     )
-
+    expected_output = """
+1 Datasource found:[0m
```
Comment (jcampbell) [inline tests/cli/test_datasource_pandas.py:67]: for my learning: I'm not familiar with this syntax...what does it mean?

File: great_expectations/cli/docs.py [hunk 4]
```diff
@@ -6,12 +6,12 @@
...
 from great_expectations import exceptions as ge_exceptions
 from great_expectations.cli.cli_logging import logger
-from great_expectations.cli.util import cli_message
+from great_expectations.cli.util import cli_message, cli_message_list
 
 
 @click.group()
 def docs():
-    """data docs operations"""
+    """Data Docs operations"""
```
Comment (Aylr) [inline great_expectations/cli/docs.py:14]: yes wonderful consistency

File: great_expectations/cli/datasource.py [hunk 5]
```diff
@@ -74,7 +77,7 @@ class SupportedDatabases(enum.Enum):
 
 @click.group()
 def datasource():
-    """datasource operations"""
+    """Datasource operations"""
```
Comment (Aylr) [inline great_expectations/cli/datasource.py:80]: yes wonderful consistency

File: great_expectations/cli/suite.py [hunk 6]
```diff
@@ -67,7 +67,7 @@ def suite():
 )
 def suite_edit(suite, datasource, directory, jupyter, batch_kwargs):
     """
-    Generate a Jupyter notebook for editing an existing expectation suite.
+    Generate a Jupyter notebook for editing an existing Expectation Suite.
```
Comment (Aylr) [inline great_expectations/cli/suite.py:70]: Love it.

File: tests/cli/test_datasource_pandas.py (hunk 3 above)
Comment (Aylr) [reply tests/cli/test_datasource_pandas.py:67]: oh, those are the ANSI color codes. I think in the future we can strip those out for tests via click runner. Side note, this is how windows users see GE.

**Review by Aylr — APPROVED** [review]
This is really a tremendous addition. Brilliant work @roblim !

**Comment by Aylr** [comment]
Tiny reminder: update the CHANGELOG please!

File: tests/cli/test_datasource_pandas.py (hunk 3 above)
Comment (roblim) [reply tests/cli/test_datasource_pandas.py:67]: Not sure exactly - I believe they're color/other formatting things.

Comment (jcampbell) [reply tests/cli/test_datasource_pandas.py:67]: 👍 Thank you!

File: great_expectations/data_context/types/base.py (hunk 2 above)
Comment (roblim) [reply great_expectations/data_context/types/base.py:193]: Couldn't figure out quick way to exclude empty keys from yaml so keeping filled in defaults for now, but nixed that default specification in instantiate_class_from_config.

**Pull request merged**